lib/ecto/migration.ex

```diff
@@ -1108,6 +1109,9 @@ defmodule Ecto.Migration do
     * `:check` - A check constraint expression. Required when creating a check constraint.
     * `:exclude` - An exclusion constraint expression. Required when creating an exclusion constraint.
     * `:prefix` - The prefix for the table.
+    * `:with` - Optional parameter for Tds adapter for create constrint.
```

Comment:
After thinking more about this, I think we should not support this option. I don't want to encourage developers to disable constraint checks. If they want to, they can write the SQL expression by hand. :) I will remove it from master!

Reply:
It make sense since if I'm not mistaken they need to enable it again explicitly after that.